doc/api/v8.md

```diff
@@ -159,6 +177,59 @@ v8.setFlagsFromString('--trace_gc');
 setTimeout(() => { v8.setFlagsFromString('--notrace_gc'); }, 60e3);
 ```
 
+## v8.writeHeapSnapshot([filename])
+<!-- YAML
+added: REPLACEME
+-->
+
+* `filename` {string} The file path where the V8 heap snapshot is to be
+  saved. If not specified, a file name with the pattern
+  `'heapdump-{pid}.{thread_id}.{sec}.{usec}.heapsnapshot'` will be
```

Comment:
The default traces are named `node_trace.${rotation}.log`, and the default reports are named `report.${yyyymmdd}.${hhmmss}.${pid}.${id++}.json`, can we have some consistency here?

Also the files should not prefixed with `heapdump`, maybe `heapsnapshot`? Though that's just repeating the file extensions but could be useful when someone sorts the file lists.

Reply:
Do you have a proposed common filename format?

Reply:
I think the report format is good enough, though it does not contain thread ids. We could just add a thread id to the mix, so maybe `${prefix}-${yyyymmdd}-${hhmmss}-${pid}-${tid}-${id++}.json`? (I prefer hyphens over dots TBH)

Also for reference, Chrome generates the profiles as `Heap-${yyyymmdd}T${hhmmss}.heapsnapshot`, the allocation profiles have the same base name but with `.heaptimeline` or `.heapprofile` as extensions. I guess we could use `Heap` or `heap` for prefixes as well.

Reply:
Ok, done. I added a utility `DiagnosticFilename` helper that can be reused. I'm not going to modify the report or event logs to use it just yet. That can be done in a separate update. The event log will be difficult to rename since there is production code out there set up to look for the existing filename pattern. Report files should be easier to modify to use the utility